dot/parachain/dispute/types/vote.go

```diff
@@ -0,0 +1,291 @@
+	return c.DisputeStatus != nil
+}
+
+func (c *CandidateVoteState) IsConfirmed() (bool, error) {
+	if c.DisputeStatus == nil {
+		return false, nil
+	}
+
+	return c.DisputeStatus.IsConfirmedConcluded()
+}
+
+func (c *CandidateVoteState) IsConcludedFor() (bool, error) {
+	if c.DisputeStatus == nil {
+		return false, nil
+	}
+
+	return c.DisputeStatus.IsConcludedFor()
+}
+
+func (c *CandidateVoteState) IsConcludedAgainst() (bool, error) {
+	if c.DisputeStatus == nil {
+		return false, nil
+	}
+
+	return c.DisputeStatus.IsConcludedAgainst()
+}
+
+// NewCandidateVoteState creates a new CandidateVoteState
+// TODO: implement this later since nothing is using it yet
+func NewCandidateVoteState(votes CandidateVotes, now uint64) (*CandidateVoteState, error) {
+	var (
+		status DisputeStatus
+		err    error
+	)
+
+	// TODO: initialize own vote state with the votes
+	ownVoteState, err := NewOwnVoteState(CannotVote{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create own vote state: %w", err)
+	}
+
+	// TODO: get number of validators
+	//numberOfValidators := 0
+
+	// TODO: get supermajority threshold
+	superMajorityThreshold := 0
+
+	isDisputed := !(len(votes.Invalid) == 0) && !(len(votes.Valid) == 0)
+	if isDisputed {
+		status, err = NewDisputeStatus()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create dispute status: %w", err)
+		}
+
+		// TODO: get byzantine threshold
+		byzantineThreshold := 0
+
+		isConfirmed := len(votes.Valid) > byzantineThreshold
+		if isConfirmed {
+			if err := status.Confirm(); err != nil {
+				return nil, fmt.Errorf("failed to confirm dispute status: %w", err)
+			}
+		}
+
+		isConcludedFor := len(votes.Valid) > superMajorityThreshold
+		if isConcludedFor {
+			if err := status.ConcludeFor(now); err != nil {
+				return nil, fmt.Errorf("failed to conclude dispute status for: %w", err)
+			}
+		}
+
+		isConcludedAgainst := len(votes.Invalid) >= superMajorityThreshold
+		if isConcludedAgainst {
+			if err := status.ConcludeAgainst(now); err != nil {
+				return nil, fmt.Errorf("failed to conclude dispute status against: %w", err)
+			}
+		}
+	}
+
+	return &CandidateVoteState{
+		Votes:         votes,
+		Own:           ownVoteState,
+		DisputeStatus: &status,
+	}, nil
+}
+
+// NewCandidateVoteStateFromReceipt creates a new CandidateVoteState from a CandidateReceipt
+func NewCandidateVoteStateFromReceipt(receipt parachainTypes.CandidateReceipt) (*CandidateVoteState, error) {
+	votes := NewCandidateVotesFromReceipt(receipt)
+	ownVoteState, err := NewOwnVoteState(CannotVote{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create own vote state: %w", err)
+	}
+
+	return &CandidateVoteState{
+		Votes: votes,
+		Own:   ownVoteState,
+	}, nil
+}
+
+// CandidateVotes is a struct containing the votes for a candidate.
+type CandidateVotes struct {
+	CandidateReceipt parachainTypes.CandidateReceipt `scale:"1"`
+	// TODO: check if we need to use btree for this in the future
```

Comment:
is this TODO resolved?

Reply:
No, not yet. I'm hoping to get a better visibility during the coordinator tests.